src/repository-status-handler.coffee

```diff
@@ -1,36 +1,44 @@
 Git = require 'git-utils'
 path = require 'path'
 
-module.exports = (repoPath, paths = []) ->
-  repo = Git.open(repoPath)
-
-  upstream = {}
-  statuses = {}
-  submodules = {}
-  branch = null
-
-  if repo?
-    # Statuses in main repo
-    workingDirectoryPath = repo.getWorkingDirectory()
-    repoStatus = (if paths.length > 0 then repo.getStatusForPaths(paths) else repo.getStatus())
-    for filePath, status of repoStatus
-      statuses[filePath] = status
-
-    # Statuses in submodules
-    for submodulePath, submoduleRepo of repo.submodules
-      submodules[submodulePath] =
-        branch: submoduleRepo.getHead()
-        upstream: submoduleRepo.getAheadBehindCount()
-
-      workingDirectoryPath = submoduleRepo.getWorkingDirectory()
-      for filePath, status of submoduleRepo.getStatus()
-        absolutePath = path.join(workingDirectoryPath, filePath)
-        # Make path relative to parent repository
-        relativePath = repo.relativize(absolutePath)
-        statuses[relativePath] = status
-
-    upstream = repo.getAheadBehindCount()
-    branch = repo.getHead()
-    repo.release()
-
-  {statuses, upstream, branch, submodules}
+process.title = 'repository-status-handler'
+
+module.exports = ->
+  @async()
+
+  process.on 'message', ({repoPath, paths}) ->
+    paths = paths or []
+
+    repo = Git.open(repoPath)
+    statuses = {}
+    upstream = {}
+    submodules = {}
+    branch = null
+
+    try
+      if repo?
```

Comment:
We check if `repo` is null, but it looks like even if it is null, we will call `.release` on it in the `finally` block. Maybe the `if` should be moved outside of the `try/finally`, like

```coffee
if repo?
  try
    # ...
  finally
    repo.release()
```

Reply:
Good catch! I didn't even realize that.

@BinaryMuse, since you opened this PR with my old changes, how do I go about making updates?